src/Components/Goods.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Goods extends React.Component {
+    state = {
+      sortMethod: '',
+    };
+
+    componentDidMount() {
+      this.defaultConditions = {...this.state};
+    }
+
+    sortFunction = (event) => {
+      const { name } = event.target;
+
+      this.setState(({ sortMethod }) => ({
+        sortMethod: sortMethod === name ? '' : name,
+      }));
+    }
+
+    reset = () => {
+      this.setState(this.defaultConditions);
+    }
+
+    render() {
+      const { goods } = this.props;
+      const { sortMethod } = this.state;
+      const visiableGoods = [...goods];
+
+      switch (sortMethod) {
+        case('sortByAlphabet'):
+        visiableGoods.sort((a, b) => a.localeCompare(b));
+        break;
+        case('sortByLength'):
+        visiableGoods.sort((a, b) => a.length - b.length);
+        break;
+        case('reverse'):
```

Comment:
I believe that there are two ways of modifying goods, one it's sorting it and another one it's reversing, so it's should be a different value in the state for `reverse`.